cpp/modmesh/math/Complex.hpp

```diff
@@ -35,52 +35,57 @@ namespace modmesh
 
 namespace detail
 {
-template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+
+template <typename T>
 struct ComplexImpl
 {
+    static_assert(std::is_floating_point_v<T>);
```

Comment:
Move the checking from the optional template type argument here, and replace with the more strict `is_floating_point_v`.  I don't think it makes sense to use integral type for complex number.